### PR TITLE
updating functionality to scan several networks

### PR DESCRIPTION
**Pull request by karthikeyan-cere**
Updated block - explorer to scan several networks.

### CODE REVIEW

File: src/modules/block-scanner/block-scanner.service.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export interface ISanitizedArgs {
 export class BlockScannerService implements BlockScannerServiceInterface {
   public logger = new Logger(BlockScannerService.name);
 
-  private api: ApiPromise;
+  private networkApi: {api: ApiPromise; type: string}[] = [];
```
Comment (evgeny-s) [inline src/modules/block-scanner/block-scanner.service.ts:45]: Let's make it plural: `networkApis`

File: src/modules/block-scanner/block-scanner.controller.ts [hunk 2]
```diff
@@ -17,31 +17,35 @@ export class BlockScannerController {
...
+    @Param('network') network: string
+  ): Promise<LatestBlockDto> {
+    return this.blockScannerService.getLatestBlock(network);
   }
 
-  @Get('balance/:address')
-  public balance(@Param('address') address: string): Promise<BalanceDto> {
-    return this.blockScannerService.getBalance(address);
+  @Get('balance/:address/:network')
+  public balance(@Param('address') address: string, @Param('network') network:string): Promise<BalanceDto> {
```
Comment (evgeny-s) [inline src/modules/block-scanner/block-scanner.controller.ts:48]: add space before `string`

File: src/modules/block-scanner/block-scanner.service.ts [hunk 3]
```diff
@@ -52,63 +52,72 @@ export class BlockScannerService implements BlockScannerServiceInterface {
     private readonly configService: ConfigService,
   ) {}
 
-  public async startScanning(): Promise<void> {
+  public async init(): Promise<any> {
     this.logger.debug('About to scan the network');
-
-    await this.init();
-    await this.processOldBlock();
+    const networks = JSON.parse(this.configService.get('NETWORKS'));
```
Comment (evgeny-s) [inline src/modules/block-scanner/block-scanner.service.ts:57]: Can we move it to the `initNetwork` function and get rid of argument?

File: src/modules/block-scanner/block-scanner.service.ts [hunk 4]
```diff
@@ -140,51 +155,60 @@ export class BlockScannerService implements BlockScannerServiceInterface {
...
     const [result, count] = await this.transactionEntityRepository.findAndCount({
-      where: {senderId: accountId},
+      where: {
+        senderId: accountId, networkType: network
+      },
       take: limit,
       skip: offset,
     });
+
+    console.log(`result: ${JSON.stringify(result)}`);
```
Comment (evgeny-s) [inline src/modules/block-scanner/block-scanner.service.ts:174]: Do we need this?
If so - replace `console.log` with `logger`

**Review by evgeny-s — CHANGES_REQUESTED** [review]
* Check comments
* Add Release note
* Shouldn't we do the appropriate changes on the FE?

File: src/modules/block-scanner/block-scanner.service.ts (hunk 3 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts:57]: I made as a separate function so in future if we want to add any custom RPC calls, we can make it here.

File: src/modules/block-scanner/block-scanner.service.ts (hunk 4 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts]: removed the console.log

File: src/modules/block-scanner/block-scanner.controller.ts (hunk 2 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.controller.ts]: added space

File: src/modules/block-scanner/block-scanner.service.ts (hunk 1 above)
Comment (karthikeyan-cere) [reply src/modules/block-scanner/block-scanner.service.ts]: renamed.

**Review by evgeny-s — APPROVED** [review]

**Pull request merged**